pkg/api/v1/defaults.go

```diff
@@ -25,232 +25,255 @@ import (
 
 func addDefaultingFuncs(scheme *runtime.Scheme) {
 	scheme.AddDefaultingFuncs(
-		func(obj *PodExecOptions) {
-			obj.Stdout = true
-			obj.Stderr = true
-		},
-		func(obj *PodAttachOptions) {
-			obj.Stdout = true
-			obj.Stderr = true
-		},
-		func(obj *ReplicationController) {
-			var labels map[string]string
-			if obj.Spec.Template != nil {
-				labels = obj.Spec.Template.Labels
-			}
-			// TODO: support templates defined elsewhere when we support them in the API
-			if labels != nil {
-				if len(obj.Spec.Selector) == 0 {
-					obj.Spec.Selector = labels
-				}
-				if len(obj.Labels) == 0 {
-					obj.Labels = labels
-				}
-			}
-			if obj.Spec.Replicas == nil {
-				obj.Spec.Replicas = new(int32)
-				*obj.Spec.Replicas = 1
-			}
-		},
-		func(obj *Volume) {
-			if util.AllPtrFieldsNil(&obj.VolumeSource) {
-				obj.VolumeSource = VolumeSource{
-					EmptyDir: &EmptyDirVolumeSource{},
-				}
-			}
-		},
-		func(obj *ContainerPort) {
-			if obj.Protocol == "" {
-				obj.Protocol = ProtocolTCP
-			}
-		},
-		func(obj *Container) {
-			if obj.ImagePullPolicy == "" {
-				// Ignore error and assume it has been validated elsewhere
-				_, tag, _ := parsers.ParseImageName(obj.Image)
+		SetDefaults_PodExecOptions,
```

Comment:
I had that originally, but there can only be one of these per package, and
the extra segment doesn't seem to add anything.

On Apr 28, 2016, at 2:25 AM, Wojciech Tyczynski <notifications@github.com>
wrote:

In pkg/api/v1/defaults.go
<https://github.com/kubernetes/kubernetes/pull/24907#discussion_r61378324>:

> -			if util.AllPtrFieldsNil(&obj.VolumeSource) {
> -				obj.VolumeSource = VolumeSource{
> -					EmptyDir: &EmptyDirVolumeSource{},
> -				}
> -			}
> -		},
> -		func(obj *ContainerPort) {
> -			if obj.Protocol == "" {
> -				obj.Protocol = ProtocolTCP
> -			}
> -		},
> -		func(obj *Container) {
> -			if obj.ImagePullPolicy == "" {
> -				// Ignore error and assume it has been validated elsewhere
> -				_, tag, _ := parsers.ParseImageName(obj.Image)
> +		SetDefaults_PodExecOptions,

@smarterclayton <https://github.com/smarterclayton> - maybe for consistency
we should have the names of those functions in exactly the same way as we
have for conversions and deep copies. So that would mean:
SetDefaults_v1_PodExectOptions

?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/24907/files/7d7f87bbd32813c76ca7e19171b635733d852fc2#r61378324>